### PR TITLE
Sync up with changes from voting branch

### PR DESCRIPTION
**Pull request by ManfredKarrer**
There have been several changes in the voting branch (and api-integration) which I want to get into master to avoid that those branches run out of sync as well as to apply the improvements.
The main changes are:
- The handling of persisted data in the p2p library (https://github.com/bisq-network/bisq-p2p/issues/8)
- Improvements of the app setup to support API projects in headless mode as well as in mixed mode
- Use custom TxBroadcaster class for better error handling 

Beside that there are many smaller trivial improvements, fixes, refactoring and cleanups.

Changes in that project:
- Add onUiReadyHandler to BisqApp and MainView
- Add startApplication and onApplicationStarted to BisqAppMain
- Improve MenuItem component
- Use TxBroadcaster.Callback() instead of FutureCallback<Transaction>()
- Apply startup changes to MainViewModel
- Apply changes of data storage

### CODE REVIEW

**Comment by ripcurlx** [comment]
ACK - checked out all branches locally and did a test trade. Everything working so far for me.

**Comment by ManfredKarrer** [comment]
So the ACK applies to all sub projects?

File: src/main/java/bisq/desktop/main/dao/wallet/send/BsqSendView.java
Comment (sqrrm) [inline src/main/java/bisq/desktop/main/dao/wallet/send/BsqSendView.java:157]: Should this be fixed now that voting branch is being merged?

Comment (ManfredKarrer) [reply src/main/java/bisq/desktop/main/dao/wallet/send/BsqSendView.java:157]: No I did not merge the dao code, just a few parts which was required due the refactoring. So the dao code in master is still the old one from a few months back...

Comment (ManfredKarrer) [reply src/main/java/bisq/desktop/main/dao/wallet/send/BsqSendView.java:157]: A main reason why not merge yet is bc of PB changes need to be tested well and if we have it in master it will be harder to change, so better for the WIP state to be more flexible with PB changes....

Comment (sqrrm) [reply src/main/java/bisq/desktop/main/dao/wallet/send/BsqSendView.java:157]: Yeah, noticing parts are there but a lot isn't.

Comment (ManfredKarrer) [reply src/main/java/bisq/desktop/main/dao/wallet/send/BsqSendView.java:157]: Yes, only technical refactorings (class rename, move) in the dao project got applied.

**Review by sqrrm — APPROVED** [review]
utACK

**Comment by ripcurlx** [comment]
@ManfredKarrer I checked out all PR's and built the client. So my testing involved the other PR's as well.

**Comment by cbeams** [comment]
Same question here as in https://github.com/bisq-network/bisq-desktop/pull/1605#issuecomment-404045248. Was it unintentional to merge this as @bisq-github-admin-2?

**Comment by ManfredKarrer** [comment]
Yes because I did not had merge rights anymore with my dev account. I merged already the other projects so it would have caused problems to leave desktop with the old version.

**Comment by cbeams** [comment]
@ManfredKarrer and I sorted out the merge permissions issue. See https://github.com/bisq-network/bisq-desktop/pull/1605#issuecomment-404095201 for details.

**Pull request merged**
